persistence/javadsl/src/test/scala/com/lightbend/lagom/javadsl/persistence/AbstractReadSideSpec.scala

```diff
@@ -108,6 +148,37 @@ trait AbstractReadSideSpec extends ImplicitSender { spec: ActorSystemSpec =>
       expectTerminated(readSide)
     }
 
+    "persisted offsets unhandled events" in {
+
+      // count = 5 from previous test steps
+      assertSelectCount("1", 5L)
+      // this is the last known offset (after processing all 5 events)
+      val offsetBefore = fetchLastOffset()
+
+      val readSide = createReadSideProcessor()
+      val p = createTestEntityRef()
+
+      p ! new TestEntity.ChangeMode(Mode.PREPEND)
+      expectMsg(new TestEntity.InPrependMode("1"))
+      p ! TestEntity.Add.of("f")
+      expectMsg(new TestEntity.Prepended("1", "f"))
+
+      // count doesn't change because ReadSide only handles Appended events
+      // InPrependMode and Prepended events are ignored
+      assertSelectCount("1", 5L)
+
+      // however, offset gets updated
+      eventually {
+        val offsetAfter = fetchLastOffset()
+        offsetBefore should not be offsetAfter
+      }
+
+      // persisted offset must have change tough
+      watch(readSide)
+      system.stop(readSide)
+      expectTerminated(readSide)
```

Comment:
Should all of this be in a `finally`? If the `eventually` fails, I guess it won't stop the read side here.